commons-hpcc/src/main/java/org/hpccsystems/commons/utils/Utils.java

```diff
@@ -190,6 +190,37 @@ public static boolean isNumeric(String str)
         return true;
     }
 
+    /**
+     * Determines if entry represented by obj does not require single quote encapsulation.
+     *
+     * @param obj
+     *            the entry
+     * @return true, if single quotes not required
+     */
+    public static boolean doesNotRequireQuotes(Object obj)
+    {
+        try
+        {
+            if (obj == null)
+                return false;
+            if (obj instanceof Number)
+                return true;
+            if (obj instanceof String)
+            {
+                if(QUOTEDSQLSTRPATTERN.matcher((String)obj).matches())
+                    return true;
+
+                USNumberFormat.parse((String)obj);
```

Comment:
It seems like we are missing something here. Are we checking to see if the string is a number?